Remove/update lines causing build failures

robfrawley opened the pull request:
This pull request amends the invalid code blocks within the documentation to fix the build errors (both locally and on `platform.sh`). ~~Additionally, this normalizes the dependency installation and build calls executed by both `travis` and `platform.sh` so they both use the same environment to build the documentation. _This is important as currently, often one will fail while the other will succeed; this should not ever happen and they should both either fail or succeed._~~

See https://github.com/symfony/symfony-docs/pull/8004#issuecomment-306661624 and https://github.com/symfony/symfony-docs/pull/8004#issuecomment-306857202 for the origin story.

~~*Note: This should likely be applied to the `3.3` branch which is where the errors originate, will update the target branch once this is confirmed.*~~

robfrawley commented:
Some interesting notes:
- Using `sphinx@1.3` allows the documentation to build **_withOUT_ error**
- Using `sphinx@1.4` prints the invalid literal blocks **_WITH_ warnings**
- Using `sphinx@1.5` stops on the invalid literal blocks **_WITH_ error**

**Sphinx 1.4 Warnings:**
```
/home/rmf/repositories/docs/symfony-docs/controller/argument_value_resolver.rst:162: WARNING: Could not lex literal_block as "xml". Highlighting skipped.                                                          
/home/rmf/repositories/docs/symfony-docs/controller/error_pages.rst:283: WARNING: Could not lex literal_block as "xml". Highlighting skipped.                                                                      
/home/rmf/repositories/docs/symfony-docs/controller/upload_file.rst:389: WARNING: Could not lex literal_block as "xml". Highlighting skipped.                                                                      
/home/rmf/repositories/docs/symfony-docs/profiler/data_collector.rst:243: WARNING: Could not lex literal_block as "xml". Highlighting skipped.                                                                     
/home/rmf/repositories/docs/symfony-docs/security/api_key_authentication.rst:432: WARNING: Could not lex literal_block as "xml". Highlighting skipped.                                                             
/home/rmf/repositories/docs/symfony-docs/service_container/configurators.rst:144: WARNING: Could not lex literal_block as "xml". Highlighting skipped.                                                             
/home/rmf/repositories/docs/symfony-docs/service_container/expression_language.rst:22: WARNING: Could not lex literal_block as "yaml". Highlighting skipped.
```

**Sphinx 1.5 Error:**
```
writing output... [ 34%] controller/argument_value_resolver                                                                                                                                                        
Exception occurred:
  File "/home/rmf/.ve/local/lib/python2.7/site-packages/sphinx/highlighting.py", line 147, in highlight_block
    type='misc', subtype='higlighting_failure')
TypeError: warner() got an unexpected keyword argument 'type'
```

So we have some options before us:
- **Fix** the documentation to build properly on newer `sphinx` versions
- **Ignore** the documentation errors and continue to use an outdated `sphinx` version

What direction do you guys want to take?

xabbuh commented:
The best option IMO would be to fix the code examples and then use Sphinx >= 1.5 to see this as soon as possible during Travis builds.

robfrawley commented:
There is one additional complication when choosing this option: warnings are thrown by `sphinx-php` when it overwrites the `code-block` directive, and these warnings are turned into errors due to the `-nW` flags passed to `sphinx` during the Travis build.

As far as I can tell, there are no alternatives for `sphinx-php` to overwrites directives, so the warning is the current expected behavior for `sphinx` when this occurs. But, this means we can't use the "turn warnings into errors" flags (`-nW`) when building the documentation on Travis.

robfrawley committed:
fix invalid code lines and remove extra space

robfrawley commented:
Rebased onto `3.3`, added changes from #8010 to align Travis and platform.sh build, and removed extra spaces from #8014.

robfrawley commented:
Rebased onto `3.3`, added a subset of the changes from #8010 to align Travis and platform.sh build so they both use identical environments, and removed extra spaces from #8014.

robfrawley commented:
Is there anything about this PR that should be changed?

xabbuh commented:
Is the `3.3` the lowest branch that contains these wrong code examples (I mean we should fix them on every branch)?

xabbuh commented:
Oh and changes to our build infrastructure should be made in the `2.7` branch to have consistent builds throughout all pull requests.

robfrawley commented:
Yes, the `3.3` branch is the lowest branch where these issues were introduced (they were mainly introduced with the autowire pull requests), short of some of the extra spaces removed, which may be in lower branches. I will open a new PR with the build changes against `2.7` and update this PR to remove them.

xabbuh commented on profiler/data_collector.rst:
```diff
@@ -254,7 +254,6 @@ to specify a tag that contains the template:
                     <tag name="data_collector"
                         template="data_collector/template.html.twig"
                         id="app.request_collector"
-                        <!-- priority="300" -->
```
this should now be added as a comment before the `tag` element

robfrawley replied:
Updated.

robfrawley commented:
Ok; split out build environment changes into #8030 and squashed commits here with requested changes.

xabbuh commented:
Thank you @robfrawley.
